core/utils/codel_test.cc

```diff
@@ -0,0 +1,288 @@
+#include "./codel.h"
+
+#include <gtest/gtest.h>
+#include <math.h>
+#include <time.h>
+#include <unistd.h>
+
+#include "../mem_alloc.h"
+#include "../packet.h"
+#include "./time.h"
+#include "./lock_free_queue.h"
+
+namespace {
+
+using bess::utils::Codel;
+using bess::utils::LockFreeQueue;
+void integer_drop(int* ptr) {
+  delete ptr;
+}
+
+// Simple test to make sure one can get back out pointer
+TEST(CodelTest, SingleInputOutput) {
+  auto q = new LockFreeQueue<Codel<LockFreeQueue, int>::Wrapper>(8);
+  Codel<LockFreeQueue, int> c(&integer_drop, q, 5000, 100000);
+  int *val = new int();
+  ASSERT_NE(val, nullptr);
+  int err;
+  c.Push(val, &err);
+  ASSERT_EQ(err, 0);
+  ASSERT_EQ(c.Size(), 1);
+  int* output = c.Pop(&err);
+  ASSERT_EQ(err, 0);
+  ASSERT_EQ(c.Size(), 0);
+  EXPECT_EQ(output, val);
+  delete val;
+}
+
+// Tests to make sure that low delay entries won't be dropped
+TEST(CodelTest, NoDropTest) {
+  auto q = new LockFreeQueue<Codel<LockFreeQueue, int>::Wrapper>(8);
+  Codel<LockFreeQueue, int> c(&integer_drop, q, 5000, 100000);
+
+  int *val1 = new int();
```

Comment:
fixed to an array of allocated integer pointers. but values need to be explicitly freed individually as some are freed by codel.